src/api/status/public/js/build-log/terminal.js

```diff
@@ -66,4 +65,28 @@ const copyToClipboard = async () => {
 };
 copyToClipboardButton.addEventListener('click', () => copyToClipboard());
 
+/**
+ * Resize the terminal depending on the height of the window
+ */
+const resizeTerminal = (function () {
+  // Since the terminal will be resized, we need to store these
+  const rect = terminal.element.getBoundingClientRect();
+  const rowHeight = rect.height;
+  const rectTop = rect.y;
+  const bottomMargin = 16;
+
+  return (windowHeight) => {
```

Comment:
Question: why do we have to create a closure instead of re-calculating everything.
```js
const resizeTerminal = function (windowHeight) {
  // Since the terminal will be resized, we need to store these
  const rect = terminal.element.getBoundingClientRect();
  const rowHeight = rect.height;
  const rectTop = rect.y;
  const bottomMargin = 16;

  const rowNumber = Math.floor((windowHeight - rectTop - bottomMargin) / rowHeight);

  terminal.resize(terminal.cols, rowNumber);
};
```

Reply:
The `terminal.element` will get updated the moment the `terminal` is resized, so if we only capture the `terminal` reference during function creation, the `y` and `rowHeight` values will change, as well.

The closure is for mainly storing the `y` and `rowHeight` so that they don't get destroyed the moment the function `resizeTerminal()` returns.